libvast/test/format/zeek.cpp

```diff
@@ -165,25 +190,33 @@ std::string_view conn_log_100_events = R"__(#separator \x09
 #close	2014-05-23-18-02-35)__";
 
 struct fixture : fixtures::deterministic_actor_system {
-  std::vector<table_slice_ptr> read(std::string_view input, size_t slice_size,
-                                    size_t num_events) {
+  std::vector<table_slice_ptr>
+  read(std::unique_ptr<std::istream> input, size_t slice_size,
+       size_t num_events, bool expect_eof) {
     using reader_type = format::zeek::reader;
     reader_type reader{defaults::system::table_slice_type, caf::settings{},
-                       std::make_unique<std::istringstream>(
-                         std::string{input})};
+                       std::move(input)};
     std::vector<table_slice_ptr> slices;
     auto add_slice = [&](table_slice_ptr ptr) {
       slices.emplace_back(std::move(ptr));
     };
     auto [err, num] = reader.read(std::numeric_limits<size_t>::max(),
                                   slice_size, add_slice);
-    if (err != ec::end_of_input)
+    if (expect_eof && err != ec::end_of_input)
```

Comment:
Slightly less code and slightly easier to read (?):
``` c++
if (err)
  if (!(expect_eof && err == ec::end_of_input))
    FAIL
```

Reply:
I think that's missing the error case where we **expect** eof but the reader returns no error.

Reply:
Does it return no error or none? 

That's an important distinction, because `if (err)` will succeed if `err` is `caf::no_error`, which is actually an error, but it won't succeed if `err` is `caf::none`, which means no error occurred.

Reply:
(We just discussed this in person and opted to adapt the error messages instead to make a bit more sense.